Stubtest: Link directly to line

Avasam opened the pull request:
This format allows editors and terminals that support linking to specific lines in files to go directly to the right line.

Before:
![image](https://user-images.githubusercontent.com/1350584/212081816-d99fa8d9-d98b-4c4b-8b3a-02dda4979cb4.png)
![image](https://user-images.githubusercontent.com/1350584/212081929-a1aa8ab3-f730-4c02-9068-b757d4e93c9d.png)

After:
![image](https://user-images.githubusercontent.com/1350584/212082039-77b08140-c780-4a45-b0b3-eb3768d8b994.png)
![image](https://user-images.githubusercontent.com/1350584/212082108-c5ae6950-9c69-44ac-b403-61515b9e2ce5.png)

AlexWaygood commented:
Looks like you'll need to update some bits of `teststubtest.py`, e.g.

https://github.com/python/mypy/blob/77f872568998bf54a24f497fef72893a60d84633/mypy/test/teststubtest.py#L1518-L1525

Avasam commented:
I didn't notice the failing CI. Thanks for letting me know. I actually went and searched through the entire codebase for any instance of "at line". And updated them (which includes docs).

I've also noticed that technically there could be a line number without a file, so I've updated for that case as well.
